packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -354,11 +355,31 @@ class _RenderCupertinoTextSelectionToolbarShape extends RenderShiftedBox {
     }
 
     final BoxParentData childParentData = child!.parentData! as BoxParentData;
+
+    final RRect rrect = _shapeRRect();
+    final Path clipPath = _clipPath(rrect);
+
+    // If configured, paint the shadow beneath the shape.
+    if (_shadowColor != null) {
```

Comment:
What's the difference between this and using BoxDecoration?

Reply:
The problem with using `BoxDecoration` to draw shadows was that you had to know upfront whether the toolbar would come above or below the anchor to apply a vertical offset. This worked with const toolbar height but not now we size it to the children.